src/pages/Proposal/ProposalSummary/index.tsx

```diff
@@ -1,15 +1,135 @@
 /** @jsxImportSource @emotion/react */
 import React from 'react';
-import { Paper } from '@mui/material';
+import { Paper, Typography } from '@mui/material';
+import { ActiveChip, BscLink, Chip, Countdown, PrimaryButton, SecondaryButton } from 'components';
+import { IProposal } from 'types';
+import { useTranslation } from 'translation';
+import Stepper from './Stepper';
 import { useStyles } from './styles';
 
-interface IProposalSummary {
+interface IProposalSummaryUiProps {
   className?: string;
+  proposal: IProposal;
 }
 
-export const ProposalSummary: React.FC<IProposalSummary> = ({ className }) => {
+interface IProposalSummaryContainerProps {
+  cancelProposal: () => void;
+  executeProposal: () => void;
+  queueProposal: () => void;
+}
+
+export const ProposalSummaryUi: React.FC<
+  IProposalSummaryUiProps & IProposalSummaryContainerProps
+> = ({ className, proposal, cancelProposal, queueProposal, executeProposal }) => {
   const styles = useStyles();
-  return <Paper css={styles.root} className={className} />;
+  const { t } = useTranslation();
+  const {
+    state,
+    id,
+    description: { title },
+    createdDate,
+    createdTxHash,
+    startDate,
+    startTxHash,
+    cancelDate,
+    cancelTxHash,
+    queuedDate,
+    queuedTxHash,
+    executedDate,
+    executedTxHash,
+    endTxHash,
+    endDate,
+  } = proposal;
+
+  let updateProposalButton = null;
+  let transactionHash = startTxHash;
+  switch (state) {
+    case 'Active':
+      updateProposalButton = (
+        <SecondaryButton onClick={cancelProposal} css={styles.updateProposalButton}>
+          {t('voteProposalUi.cancel')}
+        </SecondaryButton>
+      );
+      transactionHash = createdTxHash;
+      break;
+    case 'Canceled':
+      transactionHash = cancelTxHash;
+      break;
+    case 'Succeeded':
+      updateProposalButton = (
+        <PrimaryButton onClick={queueProposal} css={styles.updateProposalButton}>
+          {t('voteProposalUi.queue')}
+        </PrimaryButton>
+      );
+      break;
+    case 'Queued':
+      updateProposalButton = (
+        <PrimaryButton onClick={executeProposal} css={styles.updateProposalButton}>
+          {t('voteProposalUi.execute')}
+        </PrimaryButton>
+      );
+      transactionHash = queuedTxHash;
+      break;
+    case 'Defeated':
+      transactionHash = endTxHash;
+      break;
+    case 'Executed':
+      transactionHash = executedTxHash;
+      break;
+    default:
+      updateProposalButton = null;
```

Comment:
You don't need this default statement since that's already the value you assigned to `updateProposalButton`.

Reply:
I included the default case for completeness out of habit following the eslint rule `default-case`. The motivation being explicitly defining it removes doubt about whether it was forgotten or not. Happy to remove it

Reply:
Actually, turns out we also have this eslint rule because I get errors when I remove it so I'm going to leave it in.